src/services/services.ts

```diff
@@ -4316,6 +4361,159 @@ namespace ts {
+                            });
+                        }
+                    });
+
+                    // If possible, get folder completion as well
+                    if (host.getDirectories) {
+                        const directories = host.getDirectories(baseDir);
+                        forEach(directories, (d) => {
+                            const dName = getBaseFileName(removeTrailingDirectorySeparator(d));
+
+                            if (startsWith(dName, toComplete)) {
+                                result.push({
+                                    name: ensureTrailingDirectorySeparator(dName),
+                                    kind: ScriptElementKind.unknown,
+                                    kindModifiers: ScriptElementKindModifier.none,
+                                    sortText: dName
+                                });
+                            }
+                        });
+                    }
+                }
+
+                return includeExtensions ? result : deduplicate(result, (a, b) => a.name === b.name);
+            }
+
+            /**
+             * Check all of the declared modules and those in node modules. Possible sources of modules:
+             *      Modules that are found by the type checker
+             *      Modules from node_modules (i.e. those listed in package.json)
+             *          This includes all files that are found in node_modules/moduleName/ with acceptable file extensions
+             */
+            function getCompletionEntriesForNonRelativeModules(fragment: string, scriptPath: string): CompletionEntry[] {
+                return ts.map(enumeratePotentialNonRelativeModules(fragment, scriptPath), (moduleName) => {
+                    return {
+                        name: moduleName,
+                        kind: ScriptElementKind.unknown,
+                        kindModifiers: ScriptElementKindModifier.none,
+                        sortText: moduleName
+                    };
+                });
+            }
+
+            function enumeratePotentialNonRelativeModules(fragment: string, scriptPath: string) {
+                const ambientSymbolNameRegex = /^"(.+?)"$/;
+
+                // If this is a nested module, get the module name
+                const firstSeparator = fragment.indexOf(directorySeparator);
+                const moduleNameFragment = firstSeparator !== -1 ? fragment.substr(0, firstSeparator) : fragment;
+                const isNestedModule = fragment !== moduleNameFragment;
+
+                // Get modules that the type checker picked up
+                const ambientModules = ts.map(program.getTypeChecker().getAmbientModules(), (sym) => {
+                    const match = ambientSymbolNameRegex.exec(sym.name);
+                    if (match) {
+                        return match[1];
+                    }
+                    // This should never happen
+                    return sym.name;

```

Comment:
also include `@types` pacakges, these are `options.types` || `readDirectory(options.typeRoots)` || automatically included type pacakges (i.e. contents of `.\node_modules\tsconfig.json`).

@RyanCavanaugh can talk to how this works in the compiler.